webui/src/views/AddPlanModal.tsx

```diff
@@ -602,67 +712,106 @@ const RetentionPolicyView = () => {
-        </Col>
-      </Row>
+      <>
+        <Row>
+          <Col span={11}>
+            <Form.Item
+              name={["retention", "policyTimeBucketed", "yearly"]}
+              validateTrigger={["onChange", "onBlur"]}
+              initialValue={0}
+              required={false}
+            >
+              <InputNumber
+                addonBefore={<div style={{ width: "5em" }}>Yearly</div>}
+                type="number"
+              />
+            </Form.Item>
+            <Form.Item
+              name={["retention", "policyTimeBucketed", "monthly"]}
+              initialValue={0}
+              validateTrigger={["onChange", "onBlur"]}
+              required={false}
+            >
+              <InputNumber
+                addonBefore={<div style={{ width: "5em" }}>Monthly</div>}
+                type="number"
+              />
+            </Form.Item>
+            <Form.Item
+              name={["retention", "policyTimeBucketed", "weekly"]}
+              initialValue={0}
+              validateTrigger={["onChange", "onBlur"]}
+              required={false}
+            >
+              <InputNumber
+                addonBefore={<div style={{ width: "5em" }}>Weekly</div>}
+                type="number"
+              />
+            </Form.Item>
+          </Col>
+          <Col span={11} offset={1}>
+            <Form.Item
+              name={["retention", "policyTimeBucketed", "daily"]}
+              validateTrigger={["onChange", "onBlur"]}
+              initialValue={0}
+              required={false}
+            >
+              <InputNumber
+                addonBefore={<div style={{ width: "5em" }}>Daily</div>}
+                type="number"
+              />
+            </Form.Item>
+            <Form.Item
+              name={["retention", "policyTimeBucketed", "hourly"]}
+              validateTrigger={["onChange", "onBlur"]}
+              initialValue={0}
+              required={false}
+            >
+              <InputNumber
+                addonBefore={<div style={{ width: "5em" }}>Hourly</div>}
+                type="number"
+              />
+            </Form.Item>
+          </Col>
+        </Row>
+        <Form.Item
+          name={["retention", "policyTimeBucketed", "keepLastN"]}
+          label="Latest snapshots to keep regardless of age"
+          validateTrigger={["onChange", "onBlur"]}
+          initialValue={0}
+          required={cronIsSubHourly}
+          rules={[
+            {
+              validator: async (_, value) => {
+                if (cronIsSubHourly && !(value > 1)) {
+                  throw new Error("Specify a number greater than 1");
+                }
+              },
+              message: "Your schedule runs more than once per hour; choose how many snapshots to keep before handing off to the retention policy.",
+            },
+          ]}
+        >
+          <InputNumber
+            addonAfter={!schedule?.disabled && (
+              <Tooltip 
```

Comment:
nit: let's move the tooltip and associated logic into a subcomponent e.g. LastNTooltip , there's a decent amount of nuance here. 

Looks like the intent is a performance optimization-- might make sense to use something like useMemo https://react.dev/reference/react/useMemo

e.g. 

```
useMemo(() => { ... expensive calcuation ... }, [JSON.stringify(retention)]);
```